custom/enikshay/nikshay_datamigration/factory.py

```diff
@@ -0,0 +1,260 @@
+from dimagi.utils.decorators.memoized import memoized
+
+from casexml.apps.case.const import CASE_INDEX_EXTENSION
+from casexml.apps.case.mock import CaseFactory, CaseStructure, CaseIndex
+from corehq.apps.locations.models import SQLLocation
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from custom.enikshay.nikshay_datamigration.models import Outcome, Followup
+
+
+def validate_number(string_value):
+    if string_value is None or string_value.strip() == '':
+        return None
+    else:
+        return int(string_value)
+
+
+class EnikshayCaseFactory(object):
+
+    domain = None
+    patient_detail = None
+
+    def __init__(self, domain, patient_detail):
+        self.domain = domain
+        self.patient_detail = patient_detail
+        self.factory = CaseFactory(domain=domain)
+        self.case_accessor = CaseAccessors(domain)
+
+    def create_cases(self):
+        self.create_person_case()
+        self.create_occurrence_case()
+        self.create_episode_case()
+        self.create_test_cases()
+
+    def create_person_case(self):
+        person_structure = self.person()
+        person_case = self.factory.create_or_update_case(person_structure)[0]
+        person_structure.case_id = person_case.case_id
+
+    def create_occurrence_case(self):
+        if self._outcome:
+            occurrence_structure = self.occurrence(self._outcome)
+            occurrence_case = self.factory.create_or_update_case(occurrence_structure)[0]
+            occurrence_structure.case_id = occurrence_case.case_id
+
+    def create_episode_case(self):
+        if self._outcome:
+            episode_structure = self.episode(self._outcome)
+            episode_case = self.factory.create_or_update_case(episode_structure)[0]
+            episode_structure.case_id = episode_case.case_id
+
+    def create_test_cases(self):
+        if self._outcome:
+            # how many followup's do not have a corresponding outcome? how should we handle this situation?
+            self.factory.create_or_update_cases([self.test(followup) for followup in self._followups])
+
+    @memoized
+    def person(self):
+        nikshay_id = self.patient_detail.PregId
+
+        kwargs = {
+            'attrs': {
+                'case_type': 'person',
+                # 'owner_id': self._location.location_id,
```

Comment:
> how does the owner get set?

By searching for the location with the corresponding nikshay code (will uncomment this line when the locations are updated)